### ISSUE TITLE
Is this source code even legit?

### ISSUE THREAD

**Issue by riderattitude** [nic-delhi/AarogyaSetu_Android#432]
On Twitter, according to some reports this code is not an open source code.

**Comment by sreejeet** [comment]
Could you share those tweets to verify your concerns?

**Comment by riderattitude** [comment]
> Could you share those tweets to verify your concerns?

https://mobile.twitter.com/jackerhack/status/1266971326918455298

**Comment by adheep** [comment]
I agree, this code doesn't really match the real APP. Lots of 'views' are missing such as 'Home', 'Chatbot' and other important modules.

**Comment by riderattitude** [comment]
> I agree, this code doesn't really match the real APP. Lots of 'views' are missing such as 'Home', 'Chatbot' and other important modules.

So here you are its totally fake for making us busy.

**Comment by shubham201** [comment]
> I agree, this code doesn't really match the real APP. Lots of 'views' are missing such as 'Home', 'Chatbot' and other important modules.

It might not be the updated one. Did you find it completely different?

**Comment by tachyons** [comment]
Refer https://www.youtube.com/watch?v=qU16qhccknQ&feature=youtu.be by @asdofindia

**Comment by gopalindians** [comment]
> I agree, this code doesn't really match the real APP. Lots of 'views' are missing such as 'Home', 'Chatbot' and other important modules.

This app uses webview to load this url [https://web.swaraksha.gov.in/ncv19/](https://web.swaraksha.gov.in/ncv19/)

**Comment by adheep** [comment]
> > I agree, this code doesn't really match the real APP. Lots of 'views' are missing such as 'Home', 'Chatbot' and other important modules.
> 
> This app uses webview to load this url https://web.swaraksha.gov.in/ncv19/

Thanks for that, it makes sense now. 

But not sure if it fair to call it open-source, because most of the issues are on the webview for example alignment of the buttons at the bottom. But currently, we don't have an option to contribute to the majority of the features on the app.

**Comment by gopalindians** [comment]
> > > I agree, this code doesn't really match the real APP. Lots of 'views' are missing such as 'Home', 'Chatbot' and other important modules.
> > 
> > 
> > This app uses webview to load this url https://web.swaraksha.gov.in/ncv19/
> 
> Thanks for that, it makes sense now.
> 
> But not sure if it fair to call it open-source, because most of the issues are on the webview for example alignment of the buttons at the bottom. But currently, we don't have an option to contribute to the majority of the features on the app.

My opinion:
They(whosoever controls the Aarogya setu app code legally)have chosen the fastest way to track people with the help of this approach, if they choose to go by native development the development/delivery loop will be slower. That's why the webview.

They have opened the source code for security researchers, not for accepting other contributions through GitHub.
As GitHub is an non india based company, they will not open their development/release flow here, if they choose to release whatever they release will be a snapshot of code at particular point.

None of the government controlled apps were relased publicly like this previously, I think that's a good start.

I wish the open source development increases in and around the goverment apps/websites.

नमस्कारम

**Comment by vivek-owwit** [comment]
https://pib.gov.in/PressReleasePage.aspx?PRID=1626979

**Comment by gopalindians** [comment]
> https://pib.gov.in/PressReleasePage.aspx?PRID=1626979

Thanks for the link.
